help: Update for "subscribe/unsubscribe user to channel" terminology.

laurynmm opened the pull request:
Instead of referring to users being added or removed from channels, we now use subscribe or unsubscribed from channels.

[CZO discussion](https://chat.zulip.org/#narrow/channel/19-documentation/topic/add.20-.3E.20subscribe.20to.20channels.3F/near/2055413).

**Notes**:
- Renames and creates a URL redirect for the current "Add or remove users from a channel" article.
- Screenshots below do not include changes that were only updates link for the change noted above. I did manually test those links all work.
- We might want to change the left sidebar text for that article to be something shorter, see first set of screenshots below.
- Did a manual review of "add" and "remove" in the help center via a case-insensitive `git-grep` for these changes.

---

**Screenshots and screen captures:**

<details>
<summary>Archive a channel</summary>

[Current documentation](https://zulip.com/help/archive-a-channel)
![Screenshot from 2025-02-05 16-54-08](https://github.com/user-attachments/assets/495a8a29-fa04-4251-9d09-152c6e1cbeba)
</details>
<details>
<summary>Bots overview</summary>

[Current documentation](https://zulip.com/help/bots-overview)
![Screenshot from 2025-02-05 16-55-56](https://github.com/user-attachments/assets/3b882dc4-1481-4be2-9e5c-a4e133012e8f)
</details>
<details>
<summary>Channel permissions</summary>

[Current documentation](https://zulip.com/help/channel-permissions)
![Screenshot from 2025-02-05 16-57-14](https://github.com/user-attachments/assets/8c13f7ef-d5ef-4fe2-b94d-df62ad6b7019)
![Screenshot from 2025-02-05 16-57-26](https://github.com/user-attachments/assets/197990ba-c9b2-4ea1-aeb6-9c9d460318c9)
</details>
<details>
<summary>Who can administer a channel</summary>

[Current documentation](https://zulip.com/help/configure-who-can-administer-a-channel)
![Screenshot from 2025-02-05 17-00-11](https://github.com/user-attachments/assets/e0a93e19-0bfe-4820-9f70-9f7e2e081a59)
</details>
<details>
<summary>Direct messages</summary>

[Current documentation](https://zulip.com/help/direct-messages)
![Screenshot from 2025-02-05 17-02-30](https://github.com/user-attachments/assets/56ecc495-4a50-4e8f-a210-b87b999a780e)
</details>
<details>
<summary>Guest users</summary>

[Current documentation](https://zulip.com/help/guest-users)
![Screenshot from 2025-02-05 17-04-24](https://github.com/user-attachments/assets/e05aca00-5e94-4136-8e7a-8e3dfb507a27)
</details>
<details>
<summary>Create channels</summary>

[Current documentation](https://zulip.com/help/create-channels)
![Screenshot from 2025-02-05 17-06-24](https://github.com/user-attachments/assets/258db4b3-6e69-49ea-b178-b631e0f3d0c4)
</details>
<details>
<summary>Manage a user's channel subscribtions</summary>

[Current documentation](https://zulip.com/help/manage-user-channel-subscriptions)
![Screenshot from 2025-02-05 17-08-40](https://github.com/user-attachments/assets/0bea9c46-d50f-4241-abeb-80aea20233af)
![Screenshot from 2025-02-05 17-08-29](https://github.com/user-attachments/assets/b33c032c-7cea-4444-82ce-19d339027a82)
</details>
<details>
<summary>Setting Zulip up for a class</summary>

[Current documentation](https://zulip.com/help/setting-up-zulip-for-a-class)
![Screenshot from 2025-02-05 17-11-51](https://github.com/user-attachments/assets/9f25e5c6-46e8-4284-b634-ba43d8e20c74)
![Screenshot from 2025-02-05 17-12-16](https://github.com/user-attachments/assets/cd10dd51-ee04-4217-af51-cbc5282d401f)
</details>
<details>
<summary>User roles</summary>

[Current documentation](https://zulip.com/help/user-roles)
![Screenshot from 2025-02-05 17-14-10](https://github.com/user-attachments/assets/bdf8b482-b697-4fa0-8a1a-a44571648238)

</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
I think given we're changing the permissions structure to make adding and removing users from a channel separate operations, it might be the right time to split this into two articles: One on subscribing, one on unsubscribing. I imagine the redirects can go to the subscribe one. @alya seems to agree that makes sense. Do you want to rework this to take that approach?

It'll help with avoiding line wrapping in the sidebar, as well as just making many of the links to unsubscribe go to a more accurate place.

laurynmm commented:
**Updates**:
- Split article for subscribing/unsubscribing users from a channel into two separate articles: `/help/subscribe-users-to-a-channel` and `/help/unsubscribe-users-from-a-channel`.
- Related article sections that linked to that one article now link to the two new articles.
- The URL redirect for the joint article goes to the subscribe users to a channel article.
- Updated links/docs with the `add-or-remove-users...` article that were not in the help center.

---

**Updated screenshots**:

<details>
<summary>Channel management section in help center sidebar</summary>

![Screenshot from 2025-02-10 13-12-08](https://github.com/user-attachments/assets/aeced650-d184-4202-bf92-c615972e979b)
</details>
<details>
<summary>Subscribe users to a channel</summary>

![Screenshot from 2025-02-10 13-12-26](https://github.com/user-attachments/assets/76d7a3e4-2024-4253-b732-7e1d4960b3ec)
![Screenshot from 2025-02-10 13-12-42](https://github.com/user-attachments/assets/d940be4e-4e69-4338-a032-c85f62b60d4a)
![Screenshot from 2025-02-10 13-12-54](https://github.com/user-attachments/assets/8bb9c0f7-66f7-46cf-a635-d627f4b5869a)
![Screenshot from 2025-02-10 13-13-07](https://github.com/user-attachments/assets/b3672233-cc08-4a0d-bdbe-e5bdf055c5e6)
![Screenshot from 2025-02-10 13-13-18](https://github.com/user-attachments/assets/7c0de5f2-f24e-4b72-80dc-8137547ed1de)
</details>
<details>
<summary>Unsubscribe users from a channel</summary>

![Screenshot from 2025-02-10 13-13-39](https://github.com/user-attachments/assets/74b6e0a7-ba99-497b-a3b6-1309b02f3c00)
![Screenshot from 2025-02-10 13-13-52](https://github.com/user-attachments/assets/eeda00e2-8ce6-4758-81a8-c57ff2189092)
</details>

---

**Questions**:

*Do we want to reorder or change the text for one of the two 'unsubscribe' articles for this page? The second one links to the article for a user to unsubscribe themself from a channel.*

<details>
<summary>Related articles in "Manage user channel subscriptions" article</summary>

![Screenshot from 2025-02-10 13-22-52](https://github.com/user-attachments/assets/0ddc6a17-92e2-40cf-a274-dc35669f5ff5)
</details>

alya commented:
Marking for integration review. I'm getting confused trying to understand the question about related articles and don't want to pull down the PR right now, so @timabbott can tweak, or we'll notice later if something's not quite right.

timabbott commented:
I think it's fine. Merged, thanks @laurynmm!
